metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -131,21 +137,26 @@ def _configure_transforms(self) -> None:
                 )
 
     @classmethod
-    def create(cls, config_dict: dict) -> "Pipeline":
+    def create(
+        cls, config_dict: dict, dry_run: bool = False, preview_mode: bool = False
+    ) -> "Pipeline":
         config = PipelineConfig.parse_obj(config_dict)
-        return cls(config)
+        return cls(config, dry_run=dry_run, preview_mode=preview_mode)
 
     def run(self) -> None:
         callback = LoggingCallback()
         extractor: Extractor = self.extractor_class()
-        for wu in self.source.get_workunits():
+        for wu in itertools.islice(
+            self.source.get_workunits(), self.max_workunits_to_process
+        ):
             # TODO: change extractor interface
             extractor.configure({}, self.ctx)
 
             self.sink.handle_work_unit_start(wu)
```

Comment:
should this method also be protected by the `--dry-run` check?

Reply:
Nope! We can do dry-run outside of preview mode as well (where `self.max_workunits_to_process` will be `None` and we get all workunits) .